app/src/main/java/fr/neamar/kiss/dataprovider/AppProvider.java

```diff
@@ -19,22 +18,30 @@ public AppProvider(Context context) {
         ArrayList<Pojo> records = new ArrayList<>();
 
         int relevance;
-        String appNameLowerCased;
+        int matchPositionStart;
+        int matchPositionEnd;
+        String appNameNormalized;
 
-        final String highlightRegexp = "(?i)(" + StringNormalizer.unNormalize(query) + ")";
+        final String queryWithSpace = " " + query;
         for (Pojo pojo : pojos) {
             relevance = 0;
-            appNameLowerCased = pojo.nameLowerCased;
-            if (appNameLowerCased.startsWith(query))
-                relevance = 100;
-            else if (appNameLowerCased.contains(" " + query))
-                relevance = 50;
-            else if (appNameLowerCased.contains(query))
-                relevance = 1;
+            appNameNormalized = pojo.nameNormalized;
+
+            matchPositionEnd = 0;
+            if (appNameNormalized.startsWith(query)) {
+                relevance          = 100;
+                matchPositionStart = 0;
+                matchPositionEnd   = query.length();
```

Comment:
matchPositionEnd is always matchPositionStart + query.length(), maybe we can remove matchPositionEnd and send the compued value directly to setDisplayNameHighlightRegion?